### PR TITLE
Implement getTokenEndpointMap and getEndpointToHostId, fix getTokens

### LINKED ISSUE #1378: Implement getRangeToEndpointMap(String keyspace) in the HttpManagementProxy
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=37693641)

Method signature:

`Map<List<String>, List<String>> getRangeToEndpointMap(String keyspace)`

### LINKED ISSUE #1387: Implement getTokenToEndpointMap() in the HttpManagementProxy
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=37693784)

Method signature:
`public Map<String, String> getTokenToEndpointMap()`

### PR DESCRIPTION
**Pull request by emerkle826**
Fixes #1378
Fixes #1387

### CODE REVIEW

**Comment by emerkle826** [comment]
NOTE: This still needs https://github.com/k8ssandra/management-api-for-apache-cassandra/pull/377 to be merged and then a new client generated. Currently, this PR is using a client compiled from that PR branch.

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java [hunk 1]
```diff
@@ -109,19 +111,17 @@ public String getHost() {
 
   @Override
   public List<BigInteger> getTokens() {
```
Comment (Miles-Garnsey) [inline src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:114]: TODO: you need to update the test for this, as it fails currently.

Comment (emerkle826) [reply src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:114]: Yep, I missed that one. Got it fixed now

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java [hunk 2]
```diff
@@ -130,7 +130,23 @@ public List<BigInteger> getTokens() {
 
   @Override
   public Map<List<String>, List<String>> getRangeToEndpointMap(String keyspace) throws ReaperException {
-    return null; // TODO: implement me.
+    try {
```
Comment (Miles-Garnsey) [inline src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:134]: Issue: I can't see a unit test for this method.

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java [hunk 3]
```diff
@@ -431,8 +447,20 @@ public void takeColumnFamilySnapshot(String var1, String var2, String var3) thro
 
   @Override
   public Map<String, String> getTokenToEndpointMap() {
-    // TODO: implement me.
-    return new HashMap<>();
+    try {
```
Comment (Miles-Garnsey) [inline src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:452]: Issue: I can't see a unit test for this method.

**Review by Miles-Garnsey — CHANGES_REQUESTED** [review]
Apologies, I should have picked this up before, but I can't see unit tests for the two new methods. The unit test for getTokens appears to be right, but we need some for `getRangeToEndpointMap` and `getTokenToEndpointMap`

**Review by Miles-Garnsey — APPROVED** [review]

**Pull request merged**